### ISSUE TITLE
Scroll position is reset when menu is closed

### ISSUE THREAD

**Issue by houmain** [phase1geo/Annotator#70]
Whenever a menu is closed (toolbar, context, titlebar) the scrollbars are reset to origin.
I am using X11 and built from the master branch.

**Comment by phase1geo** [comment]
I can replicate this behavior on my end (I'm running on Fedora with Wayland -- so I don't think the issue is only with X11).  However, I can't find any code in my codebase that is controlling this behavior.  I did notice that if I move the scrollbars, click on the Open button in the header bar to reveal its menu, and then click the Open button again to clear the menu, I could click on other buttons with menus without triggering the scrollbars to change their reported position.  I'm not suggesting this as a workaround because as soon as you click into the image, the scrollbars jump again.

I'll keep looking into this, but I'm not sure what is causing the issue at the moment.

**Comment by houmain** [comment]
Hi, I tried to track this down and identified the `grab_focus()` as the cause.

I found out that its behavior changes between Gtk 3.x and 4:
https://discourse.gnome.org/t/disable-auto-scrolling-in-gtkscrolledwindow-when-grab-focus-in-children/13058

I could not find a solution/workaround other than overriding `Canvas.grab_focus` with an empty function... (but I don't know anything about Vala/Gtk).

I have Gtk 3.24.35 installed.
